server/router/src/main/java/org/infinispan/server/router/RoutingTable.java

```diff
@@ -0,0 +1,44 @@
+package org.infinispan.server.router;
+
+import org.infinispan.server.router.routes.Route;
+import org.infinispan.server.router.routes.RouteDestination;
+import org.infinispan.server.router.routes.RouteSource;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class RoutingTable {
+
+    private Set<Route<? extends RouteSource, ? extends RouteDestination>> routes = new HashSet<>();
+
+    public RoutingTable(Set<Route<? extends RouteSource, ? extends RouteDestination>> routes) {
+        this.routes.addAll(routes);
+    }
+
+    public int routesCount() {
+        return routes.size();
+    }
+
+    public Stream<Route<? extends RouteSource, ? extends RouteDestination>> streamRoutes() {
```

Comment:
Not sure I understand the need to expose a Stream here. If routes are static and cannot be modified at runtime and the size is known, a simple array could be exposed? If exposing an array does not work well, you could just wrap it around an immutable List which is much easier for the caller to work with. A stream is good for transformation but if the caller is simply going to iterate with a `forEach`, I don't see much value.